pkg/controller/jobframework/reconciler.go

```diff
@@ -820,7 +820,13 @@ func (r *JobReconciler) stopJob(ctx context.Context, job GenericJob, wl *kueue.W
 	}
 
 	if jws, implements := job.(ComposableJob); implements {
-		stoppedNow, err := jws.Stop(ctx, r.client, info, stopReason, eventMsg)
+		reason := stopReason
+		if stopReason == StopReasonWorkloadEvicted {
+			if evCond := apimeta.FindStatusCondition(wl.Status.Conditions, kueue.WorkloadEvicted); evCond != nil && evCond.Status == metav1.ConditionTrue {
```

Comment:
You can use `apimeta.IsStatusConditionTrue`

Reply:
Right, but I do need the entire `Condition` object (not only its presence/absence) because I have to read the `Reason` from it.

Reply:
oh, yes